File: 2659-make-array-empty/2659-make-array-empty.py
class Solution:
    def countOperationsToEmptyArray(self, nums: List[int]) -> int:
        n = len(nums)
        pos = sorted(range(n), key=lambda i: nums[i])
        return n + sum(n - i for i in range(1, n) if pos[i] < pos[i - 1])

File: 2659-make-array-empty/2659-make-array-empty.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countOperationsToEmptyArray = function(nums) {
  const pos = nums.map((_, i) => i).sort((a, b) => nums[a] - nums[b]);
  const n = nums.length;
  let ans = n;
  
  for (let i = 1; i < n; i++) {
    if (pos[i] < pos[i - 1]) ans += n - i;
  }
    
  return ans;
};